IndicoV2/Jobs/JobsClient.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using IndicoV2.StrawberryShake;
+using IndicoV2.Jobs.Exceptions;
+using Newtonsoft.Json.Linq;
+using JobStatus = IndicoV2.Jobs.Models.JobStatus;
+
+namespace IndicoV2.Jobs
+{
+    public class JobsClient : IJobsClient
+    {
+        private readonly IndicoStrawberryShakeClient _strawberryShake;
+        private readonly IndicoClient _indicoClient;
+
+        private readonly JobStatus[] _waitingForResult = {
+            JobStatus.PENDING,
+            JobStatus.RECEIVED,
+            JobStatus.STARTED,
+        };
+
+        public JobsClient(IndicoClient indicoClient)
+        {
+            _indicoClient = indicoClient;
+            _strawberryShake = indicoClient.IndicoStrawberryShakeClient;
+        }
+
+        public async Task<JobStatus> GetStatusAsync(string jobId, CancellationToken cancellationToken = default)
+        {
+            var result = await _strawberryShake.Jobs().GetStatusAsync(jobId, cancellationToken);
+            return (JobStatus)result.Job.Status.Value;
+        }
+
+        public async Task<string> GetResultAsync(string jobId, TimeSpan checkInterval, CancellationToken cancellationToken = default)
+        {
+            JobStatus status;
+            while (_waitingForResult.Contains(status = await GetStatusAsync(jobId, cancellationToken)))
+            {
+                await Task.Delay(checkInterval, cancellationToken);
+            }
+
+            if (status != JobStatus.SUCCESS)
+            {
+                var failReason = await GetFailureReasonAsync(jobId);
+                throw new JobNotSuccessfulException(status, failReason);
+            }
+
+            var result = await _strawberryShake.Jobs().GetResultAsync(jobId, cancellationToken);
+            return result.Job.Result;
```

Comment:
`var json = JsonConvert.DeserializeObject<TResult>(result);` is what we were doing in the previous iteration that let us keep this as a generic function. We should do something like that again here.

Reply:
cool thanks!